src/aliceVision/calibration/checkerDetector.cpp

```diff
@@ -1308,24 +1379,556 @@ bool CheckerDetector::mergeCheckerboards()
+
+        if (votes.find(offset) == votes.end())
+        {
+            votes[offset] = 0;
+        }
+        else
+        {
+            votes[offset]++;
+        }
+    }
+
+    // Find the best vote
+    int max_count = 0;
+    Vec2 max_offset;
+    for (auto& v : votes)
+    {
+        if (v.second > max_count)
+        {
+            max_count = v.second;
+            max_offset.x() = v.first.first;
+            max_offset.y() = v.first.second;
+        }
+    }
+
+    if (max_count < 5)
+    {
+        return false;
+    }
+
+    // If needed, double the current checkerboard size to give the same discretization of space
+    if (scale > 1)
+    {
+        CheckerBoard double_current_board(cur_board.rows() * scale, cur_board.cols() * scale);
+
+        double_current_board.fill(UndefinedIndexT);
+
+        for (int i = 0; i < cur_board.rows(); i++)
+        {
+            for (int j = 0; j < cur_board.cols(); j++)
+            {
+                double_current_board(i * scale, j * scale) = cur_board(i, j);
+            }
+        }
+
+        cur_board = double_current_board;
+    }
+
+    int cur_board_left = max_offset.x();
+    int cur_board_top = max_offset.y();
+    int ref_board_left = 0;
+    int ref_board_top = 0;
+
+    if (cur_board_left < 0)
+    {
+        cur_board_left = 0;
+        ref_board_left = -max_offset.x();
+    }
+
+    if (cur_board_top < 0)
+    {
+        cur_board_top = 0;
+        ref_board_top = -max_offset.y();
+    }
+
+    int width_output = std::max(cur_board_left + cur_board.cols(), ref_board_left + ref_board.cols());
+    int height_output = std::max(cur_board_top + cur_board.rows(), ref_board_top + ref_board.rows());
+
+    CheckerBoard output(height_output, width_output);
+    output.fill(UndefinedIndexT);
+
+    ref_center.x() += ref_board_left;
+    ref_center.y() += ref_board_top;
+
+    for (int i = 0; i < ref_board.rows(); i++)
+    {
+        for (int j = 0; j < ref_board.cols(); j++)
+        {
+            output(ref_board_top + i, ref_board_left + j) = ref_board(i, j);
+        }
+    }
+
+    for (int i = 0; i < cur_board.rows(); i++)
+    {
+        for (int j = 0; j < cur_board.cols(); j++)
+        {
+            IndexT id = cur_board(i, j);
+            if (id != UndefinedIndexT)
+            {
+                output(cur_board_top + i, cur_board_left + j) = cur_board(i, j);
+            }
+        }
+    }
+
+    std::vector<CheckerBoard> output_boards;
+    output_boards.push_back(output);
+
+    for (int idx = 1; idx < _boards.size(); idx++)
+    {
+        if (idx != other)
+        {
+            output_boards.push_back(_boards[idx]);
+        }
+    }
+
+    _boards = output_boards;
+
+    return true;
+}
+
+void CheckerDetector::groupNestedCheckerboards()
+{
+    int current_scale = 1;
+    bool change = true;
+
+    if (_boards.size() == 0)
+    {
+        return;
+    }
+
+    Vec2i center = { _boards[0].cols() / 2, _boards[0].rows() / 2 };
```

Comment:
```suggestion
    const Vec2i center = { _boards[0].cols() / 2, _boards[0].rows() / 2 };
```

Reply:
Actually this one cannot be const as it is passed by reference to `groupNestedCheckboardsPair` which is modifying its value